torch/distributed/tensor/parallel/_utils.py

```diff
@@ -46,15 +48,53 @@ def _validate_tp_mesh_dim(
 
     parent_mesh = _mesh_resources.get_parent_mesh(device_mesh)
     if parent_mesh:
-        if parent_mesh.ndim != 2:
-            raise RuntimeError(
-                f"Found TP device_mesh has a parent mesh with dims {parent_mesh.ndim}",
-                "Currently we only support 2D TP composition with DP.",
-            )
+        # if parent_mesh.ndim != 2:
```

Comment:
I think it's fine to remove this constraint, this is mainly because we only have a 2-D checkpoint save/load enabled in dcp at that time, for 3D, we can test and verify the save/load works separately

Reply:
cc @wz337